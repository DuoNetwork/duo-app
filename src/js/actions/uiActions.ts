import * as CST from '../common/constants';
import * as reduxTypes from '../common/reduxTypes';

export function refresh(): reduxTypes.Action {
	return {
		type: CST.AC_REFRESH
	};
}

export function addHistory(tx: string): reduxTypes.Action {
	return {
		type: CST.AC_HISTORY,
		value: tx
	};
}
