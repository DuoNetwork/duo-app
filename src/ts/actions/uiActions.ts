import * as CST from 'ts/common/constants';

export function updatePeriod(period: number) {
	return { type: CST.AC_UI_PERIOD, value: period };
}

export function updateSource(src: string) {
	return {
		type: CST.AC_UI_SOURCE,
		value: src
	};
}

export function localeUpdate(locale: string) {
	return {
		type: CST.AC_UI_LOCALE,
		value: locale
	};
}
