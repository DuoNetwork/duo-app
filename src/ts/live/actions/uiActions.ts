import * as CST from '../common/constants';
import * as reduxTypes from '../common/reduxTypes';
import * as contractActions from './contractActions';
import * as dynamoActions from './dynamoActions';

export function refreshUpdate(): reduxTypes.Action {
	return {
		type: CST.AC_REFRESH
	};
}
export function refresh(): reduxTypes.ThunkAction {
	return async dispatch => {
		dispatch(contractActions.getCustodianStates());
		dispatch(contractActions.getCustodianPrices());
		dispatch(contractActions.getBalances());
		dispatch(dynamoActions.fetchHourly());
		dispatch(dynamoActions.fetchMinutely());
		dispatch(refreshUpdate());
	};
}
