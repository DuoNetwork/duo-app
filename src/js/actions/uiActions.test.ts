//import configureMockStore from 'redux-mock-store';
//import thunk from 'redux-thunk';
//import * as CST from '../common/constants';
import * as uiActions from './uiActions';

//const mockStore = configureMockStore([thunk]);

describe('actions', () => {
	test('refresh', () => {
		expect(uiActions.refresh()).toMatchSnapshot();
	});

	test('addHistory', () => {
		expect(uiActions.addHistory('test')).toMatchSnapshot();
	});
});
