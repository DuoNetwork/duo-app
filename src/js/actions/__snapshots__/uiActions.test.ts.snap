// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions addHistory 1`] = `
Object {
  "type": "history",
  "value": "test",
}
`;

exports[`actions refresh 1`] = `
Object {
  "type": "refresh",
}
`;
